## `Quantis` e `quartis` nos ajudam a entender o que ocorre em certa porcentagem dos nossos dados. Eles recortam nossos dados na quantidade que pedimos, e então temos uma noção melhor de como as observações estão se comportando.
<br>

![alt text](https://academy.vertabelo.com/static/python-data-science-04-quartiles-3.png)

### O percentil 10% (quantil 0.10), por exemplo, nos informa qual é o valor máximo dos dados se eu considerar somente os 10% primeiros termos dos meus dados, depois de ordená-los. 

## Os quartis são quantis específicos: 

#### * 1˚ Quartil = Quantil 0.25
#### * 2˚ Quartil = Mediana = Quantil 0.50
#### * 3˚ Quartil = Quantil 0.75

### Vamos continuar usando os dados do preço do Big Mac de Julho de 2022.

<br>

https://www.kaggle.com/datasets/vittoriogiatti/bigmacprice

#### Importando a biblioteca pandas (https://pandas.pydata.org)

In [1]:
import pandas

#### Lendo os dados, que estão em csv

In [2]:
df = pandas.read_csv('BigmacPriceJuly2022.csv')

In [3]:
df.head()

date currency_code        name  local_price  dollar_ex  dollar_price
0  2022-07-01           ARS   Argentina       590.00        129          4.57
1  2022-07-01           AUD   Australia         6.70          1          6.70
2  2022-07-01           EUR     Austria         4.35          1          4.35
3  2022-07-01           AZN  Azerbaijan         4.70          1          4.70
4  2022-07-01           BHD     Bahrain         1.60          1          1.60

### Descrição das colunas

* **`date`** - Data do registro
* **`currency_code`** - Código ISO 4217 internacional da moeda do país
* **`name`** - Nome do país
* **`local_price`** - Preço do Big Mac na moeda local
* **`dollar_ex`** - Taxa de Conversão do dólar americano para a moeda local
* **`local_price`** - Preço do Big Mac em dólares americanos


### Vamos calcular alguns quantis para o  preço do Big Mac em dólares americanos

Aqui calculamos o quantil 0.1, ou o percentil 10%

In [4]:
df['dollar_price'].quantile(0.1)

2.413

Podemos afirmar que 10% dos Big Mac's do mundo custam até 2.413 dólares

Aqui calculamos o quantil 0.25, ou o percentil 25%, ou o primeiro quartil

In [6]:
df['dollar_price'].quantile(0.25)

3.0

Podemos afirmar que um quarto dos Big Mac's do mundo custam até 3 dólares

Aqui calculamos o quantil 0.5, ou o percentil 50%, ou a mediana

In [8]:
df['dollar_price'].quantile(0.5)

3.9050000000000002

In [9]:
df['dollar_price'].median()

3.9050000000000002

Podemos dizer que metade dos Big Mac's do mundo custam até 3.9 dólares

### Agora com os quantis, conseguimos montar um retrato mais completo de como nossos dados se comportam.

In [13]:
percentil_10 = df['dollar_price'].quantile(0.1)

percentil_25 = df['dollar_price'].quantile(0.25)

percentil_50 = df['dollar_price'].quantile(0.50)

percentil_75 = df['dollar_price'].quantile(0.75)

percentil_90 = df['dollar_price'].quantile(0.9)

percentil_95 = df['dollar_price'].quantile(0.95)

percentil_99 = df['dollar_price'].quantile(0.99)

media = df['dollar_price'].mean()

moda = df['dollar_price'].mode()[0]

In [14]:
print("Quantil 0.10: {}\n".format(percentil_10))

print("Primeiro Quartil: {}\n".format(percentil_25))

print("Mediana: {}\n".format(percentil_50))
print("Média: {}\n".format(media))
print("Moda: {}\n".format(moda))

print("Terceiro Quartil: {}\n".format(percentil_75))

print("Quantil 0.90: {}\n".format(percentil_90))

print("Quantil 0.95: {}\n".format(percentil_95))

print("Quantil 0.99: {}\n".format(percentil_99))

Quantil 0.10: 2.413

Primeiro Quartil: 3.0

Mediana: 3.9050000000000002

Média: 4.003571428571429

Moda: 4.0

Terceiro Quartil: 4.6875

Quantil 0.90: 5.91

Quantil 0.95: 6.609999999999999

Quantil 0.99: 6.9551



### Os quantis são de muita importância quando queremos entender melhor a própria dispersão dos dados que estamos analisando. Alguns exemplos dentre as várias aplicações para eles:

* Identificar comportamentos muito diferentes do restante dos dados (Outliers)
* Entender em qual corte a maioria dos dados se encontram, ou até mesmo uma parte específica deles
* Ajudar na tomada de decisão em situações de Trade-Off. Ex: Acurácia de Localização
* Boxplot